Add my contributions to release note highlights (#4076)
ElektraInitiative/libelektra: doc/news/_preparation_next_release.md

## Highlights

- `kdb` now prohibits write operations on cascading keys that miss a corresponding existing key. See the details in the `Tools` section below and the new subsection on cascading writes in the [tutorial](/doc/tutorials/cascading.md) on cascading keys for further information. _(Alexander Firbas)_
- <<HIGHLIGHT2>>
- <<HIGHLIGHT3>>

### Redshift and Elektra

We have created a version of [Redshift](https://github.com/jonls/redshift/) that uses Elektra for configuration management!

> Redshift's code for loading configuration files, parsing CLI options and validating configuration was refactored to use Elektra.

Reviewer: Verbs should be early and also state our goal: that apps have less code for config stuff.

```suggestion
We removed and refactored Redshift's code for loading configuration files, parsing CLI options and validating configuration to use Elektra.
```

Author: Thanks, I've adopted and extended your suggestion!

> To test it, clone the [elektrify branch](https://github.com/qwepoizt/redshift/tree/improve-config/elektrify) and follow the instructions provided in [CONTRIBUTING.md](https://github.com/qwepoizt/redshift/blob/improve-config/elektrify/CONTRIBUTING.md).

Author: Sorry, I don't understand. Are you proposing we link: 
1. to redshift's PR 837 instead of the elektrify branch and 
2. to https://github.com/jonls/redshift/blob/632fb69c0d7c60b9b18ec87430e54e3f582c8eff/CONTRIBUTING.md instead of https://github.com/qwepoizt/redshift/blob/improve-config/elektrify/CONTRIBUTING.md
?

> To test it, clone the [elektrify branch](https://github.com/qwepoizt/redshift/tree/improve-config/elektrify) and follow the instructions provided in [CONTRIBUTING.md](https://github.com/qwepoizt/redshift/blob/improve-config/elektrify/CONTRIBUTING.md).

Reviewer: > 1. to redshift's PR 837 instead of the elektrify branch

Yes, to the PR 837, so that people leave comments at the right place.

> To test it, clone the [elektrify branch](https://github.com/qwepoizt/redshift/tree/improve-config/elektrify) and follow the instructions provided in [CONTRIBUTING.md](https://github.com/qwepoizt/redshift/blob/improve-config/elektrify/CONTRIBUTING.md).

Author: Good idea, thanks, I've updated that accordingly.

> 

Reviewer: No description of what is now different in redshift with Elektra.

Author: I've added a new section for this.

### HL API improvements

We have made a large number of improvements to the high-level API. The highlights are:

- improved detection of differences in specification between an application's compilation and runtime
- improved validation of CLI options
- early detection of errors in specification files
- updated and improved documentation and tutorials
- bugfixes

More improvements and details are explained below.

Thanks to _Klemens Böswirth_, _Markus Raab_ and _Tobias Schubert_!

### Windows releases

> We are now shipping experimental releases for Windows 32- and 64-bit! They can be downloaded [here](https://build.libelektra.org/job/libelektra-release/lastSuccessfulBuild/artifact/artifacts/).

Reviewer: Also write about the success that redshift works in Windows.

Author: Good idea!

## Highlights

- `kdb` now prohibits write operations on cascading keys that miss a corresponding existing key. See the details in the `Tools` section below and the new subsection on cascading writes in the [tutorial](/doc/tutorials/cascading.md) on cascading keys for further information. _(Alexander Firbas)_
- <<HIGHLIGHT2>>
- <<HIGHLIGHT3>>

### Redshift and Elektra

We have created a version of [Redshift](https://github.com/jonls/redshift/) that uses Elektra for configuration management!

We removed and refactored Redshift's code for loading configuration files, parsing CLI options and validating configuration to use Elektra. Redshift with Elektra has nearly 700 fewer lines of code (-17%) and is a great example of what Elektra is all about: More applications with less code for configuration management!

To test it, clone the [elektrify branch](https://github.com/qwepoizt/redshift/tree/improve-config/elektrify) and follow the instructions provided in [CONTRIBUTING.md](https://github.com/qwepoizt/redshift/blob/improve-config/elektrify/CONTRIBUTING.md).

#### Benefits of Redshift using Elektra

Refactoring Redshift to use Elektra brings the following benefits:

> - Fewer lines of code (~700 less LOC, 17%).

Reviewer: At least once you should mention that the 17% is about configuration related code, not the total code.

Author: Thanks for pointing that out. I've made the calculation more transparent and updated the numbers accordingly.

> - Adding new configuration settings is easier: Validation and parsing of setting values (from configuration file and CLI options) is handled by Elektra - no custom code required!

Reviewer: Maybe write also about the time spent?

Author: Done!

- Clean separation of application code and the specification of supported configuration settings (including defaults and validation rules).
- Automatic generation of CLI help text.

### HL API improvements

> We have made a large number of improvements to the high-level API. The highlights are:

Reviewer: Link to (tutorial of) HL API and also say that redshift uses it.

Author: Good idea, thanks!

- improved detection of differences in specification between an application's compilation and runtime
- improved validation of CLI options
- early detection of errors in specification files
- updated and improved documentation and tutorials

> - bugfixes

Reviewer: ```suggestion
- and various other bugfixes
```

Author: Fixed.

> More improvements and details are explained below.

Reviewer: ```suggestion
More improvements and details are explained in later sections of these release notes.
```

Author: Thanks, done!
